pkg/3scale/amp/operator/apicast_reconciler.go

```diff
@@ -545,6 +555,69 @@ func apicastPodTemplateEnvConfigMapAnnotationsMutator(desired, existing *appsv1.
 	return updated, nil
 }
 
+func (r *ApicastReconciler) reconcileAPImanagerCR(ctx context.Context) (ctrl.Result, error) {
+
+	changed := false
+	changed, err := r.reconcileApimanagerSecretLabels(ctx)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	if changed {
+		err = r.Client().Update(ctx, r.apiManager)
+		r.logger.Info("reconciling", "error", err)
+	}
+
+	return ctrl.Result{Requeue: changed}, err
```

Comment:
Although that's not a big deal I would return nil instead of empty err to keep in line with what we are doing in other return ctrl.Result, error

Reply:
Done, Thank you @MStokluska . Sorry, I did at same commit (amend).  
Retested.